vital/algo/serialize_metadata.h

```diff
@@ -0,0 +1,134 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief Interface for serialize_metadata \link
+ * kwiver::vital::algo::algorithm_def algorithm
+ *        definition \endlink.
+ */
+
+#ifndef VITAL_ALGO_SERIALIZE_METADATA_H_
+#define VITAL_ALGO_SERIALIZE_METADATA_H_
+
+#include <vital/vital_config.h>
+
+#include <string>
+
+#include <vital/algo/algorithm.h>
+#include <vital/algorithm_capabilities.h>
+#include <vital/types/metadata.h>
+#include <vital/types/metadata_map.h>
+
+namespace kwiver {
+
+namespace vital {
+
+namespace algo {
+
+/// An abstract base class for reading and writing meatadata maps
+
+/**
```

Comment:
We might take this opportunity to switch to the new standard for KWIVER doxygen where we use only `///` for each line and not `/**`.

Reply:
Yup, noted also in my review.